src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2661,25 +2682,32 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
             else:
                 site_config.app_settings.append(NameValuePair(name='WEBSITES_ENABLE_APP_SERVICE_STORAGE',
                                                               value='true'))
-                if runtime not in FUNCTIONS_VERSION_TO_SUPPORTED_RUNTIME_VERSIONS[functions_version]:
-                    raise CLIError("An appropriate linux image for runtime:'{}', "
-                                   "functions_version: '{}' was not found".format(runtime, functions_version))
         if deployment_container_image_name is None:
-            site_config.linux_fx_version = _get_linux_fx_functionapp(functions_version, runtime, runtime_version)
+            site_config.linux_fx_version = site_config_json['linuxFxVersion']
     else:
         functionapp_def.kind = 'functionapp'
-        if runtime == "java":
-            site_config.java_version = _get_java_version_functionapp(functions_version, runtime_version)
+        if runtime == "java" and 'JavaVersion' in site_config_json:
+            site_config.java_version = site_config_json['JavaVersion']
+        elif runtime == "powershell" and 'PowerShellVersion' in site_config_json:
+            site_config.power_shell_version = site_config_json['PowerShellVersion']
```

Comment:
Also, is there a way we can avoid hard coding keys such as `PowerShellVersion` ? Long term goal is, when functions adds support to new languages or versions all we have to do is update the stacks.json content without need to update any code. Is this possible?